seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java

```diff
@@ -273,12 +300,42 @@ private void initializeWithRemoteUserLogin() throws Exception {
                         configuration,
                         hadoopConf.getRemoteUser(),
                         (configuration, userGroupInformation) -> {
-                            final FileSystem fileSystem = FileSystem.get(configuration);
+                            this.userGroupInformation = userGroupInformation;
+                            this.fileSystem = FileSystem.get(configuration);
                             return Pair.of(userGroupInformation, fileSystem);
                         });
         log.info("Create FileSystem success with RemoteUser: {}.", hadoopConf.getRemoteUser());
-        this.userGroupInformation = pair.getKey();
-        this.fileSystem = pair.getValue();
-        this.fileSystem.setWriteChecksum(false);
+        userGroupInformation = pair.getKey();
+        fileSystem = pair.getValue();
+        fileSystem.setWriteChecksum(false);
+    }
+
+    private <T> T execute(PrivilegedExceptionAction<T> action) throws IOException {
```

Comment:
Please add some comments about why we need execute method to run action?

Reply:
Thanks, added the necessary comments.